### DISCUSSION TITLE
v.0.12.2

### DISCUSSION THREAD

**Discussion by DavidLazarescu** [Librum-Reader/Librum#160] [Announcements]
A big release with a lot of major changes!

**Features:**
- Added Tools Page
    - Merge
    - Extract
    - Image to PDF
- Added alerts to improve feedback
- Added occasional feedback popup
- Reworked the selection menu to look more modern

**Improvements:**
- Improved loading animation across the application and added loading animation to more places to improve feedback
- Improved 'Save To Files' to correctly use the title and file extension as destination name
- Improved feedback when downloading a book by adding a loading animation
- Remember the folder you were in after leaving the home page and restore it correctly

**Translations:**
- Added Bengali translation

**Fixes:**
- Fixed bug causing being unable to download and read books other than pdf and epub from the cloud
- Fixed bug causing crash and locally uninstalling book when losing internet connection while reading
- Fixed crash caused by uninstalling or deleting a book while its downloading
- Fixed bug causing unselected books in multi-select mode to still be logically selected and operated upon
- Fixed book title in external book mode not being displayed
- Fixed zooming not working in external book mode
- Fixed file dialog issues
- Fixed issues with incorrect current page count in books with different page heights
- Fixed issues with the zoom combobox on the reading page

<hr /><em>This discussion was created from the release <a href='https://github.com/Librum-Reader/Librum/releases/tag/v.0.12.2'>v.0.12.2</a>.</em>

**Comment by Vargock** [comment]
Thanks for all the hard work you've been doing — it's a really great app and you've been doing an amazing job with it!

**Comment by G0N2a1** [comment]
Any news for android version?
